src/hotspot/share/compiler/compilationPolicy.cpp

```diff
@@ -907,6 +907,12 @@ CompileTask* CompilationPolicy::select_task(CompileQueue* compile_queue, JavaThr
       continue;
     }
     Method* method = task->method();
+    if (task->is_scc()) {
+      // AOT task, it should load fast, take it, and go.
+      max_task = task;
+      max_method = method;
+      break;
```

Comment:
I think we can just return `task` here.

We don't cache `CompLevel_full_profile` Level3 C1 compiled code and as result we do nothing in the rest of this method for AOT code.

We put SCC tasks on separate queues which will have only these tasks.

Reply:
Right, that makes sense. Since we want to get the SC tasks as soon as possible, it makes sense to shortcut hard. I'll do it in new commit.